frontend/public/components/factory/modal.tsx

```diff
@@ -82,10 +82,10 @@ export const ModalSubmitFooter: React.SFC<ModalSubmitFooterProps> = ({message, e
 
   return <ModalFooter inProgress={inProgress} errorMessage={errorMessage} message={message}>
     <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
-      <Button type="button" variant="secondary" onClick={onCancelClick}>{cancelText || 'Cancel'}</Button>
+      <Button type="button" variant="secondary" data-test-id="cancel-action" onClick={onCancelClick}>{cancelText || 'Cancel'}</Button>
       {submitDanger
-        ? <Button type="submit" variant="danger" isDisabled={submitDisabled} id="confirm-action">{submitText}</Button>
-        : <Button type="submit" variant="primary" isDisabled={submitDisabled} id="confirm-action">{submitText}</Button>}
+        ? <Button type="submit" variant="danger" isDisabled={submitDisabled} id="confirm-action" data-test-id="confirm-action">{submitText}</Button>
+        : <Button type="submit" variant="primary" isDisabled={submitDisabled} id="confirm-action" data-test-id="confirm-action">{submitText}</Button>}
```

Comment:
We should consider using `data-test-id` instead of `#confirm-action` in other tests. That can be a follow on, though.